### ISSUE TITLE
Plugin does not work with SPA Paths

### ISSUE THREAD

**Issue by theaccordance** [driftyco/ionic-plugin-deeplinks#3]
Trying to switch over to using this plugin over `cordova-universal-links-plugin` since it's more backwards compatible, but I can't get the plugin to work when my `apple-site-association` file contains a single-page-application path: `/public/#/ResetPassword/*`.  If I change the path to a wildcard `*`, then I'm able to tap a link in iOS and have it open in app.  

The SPA path works with `cordova-universal-links-plugin`, albiet with a [slight caveat](https://github.com/nordnet/cordova-universal-links-plugin/issues/29)

**Comment by theaccordance** [comment]
Any suggestions on how to negate this?

**Comment by mlynch** [comment]
Hey @theaccordance, could you publish your apple-app-site-association file, or email it to me? max AT ionic DOT io. I can test against it, thanks.

**Comment by mlynch** [comment]
Should be all fixed up. You'll need to have the hash in your route list to match. Thanks!

**Comment by theaccordance** [comment]
Thanks Max!

On Wed, Jun 29, 2016 at 3:29 PM, Max Lynch <notifications@github.com> wrote:

> Should be all fixed up. You'll need to have the hash in your route list to
> match. Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/driftyco/ionic-plugin-deeplinks/issues/3#issuecomment-229478255>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABuqCXMKZRIjiERqqijs2DzRlEAEJRFYks5qQtWRgaJpZM4I8Bca>
> .
>



-- 
Joe Mainwaring
Email: jm@theaccordance.com
Twitter: @theaccordance
http://joemainwaring.com

**Comment by theaccordance** [comment]
Hey @mlynch, looks like this issue isn't resolved.  When I first tested this after your patch, my test server's association file wasn't properly scoped (path was still to `*` instead of `/public/#/ResetPassword/*`). I know your patch focused more on the JS side of things, but I'm wondering if this is a ObjC/Swift issue instead?

When I have the path set to `/public/#/ResetPassword/*` in the association file and tap the corresponding link on the device, it doesn't attempt to launch the app at all.  Looking at the Output logger in XCode, the only reference to the plugin I see is that it was successfully loaded:

`2016-07-05 15:38:23.533 HGDemo[2644:872425] [CDVTimer][ionicdeeplinkplugin] 0.203013ms`

However, if I have the path set to `*` in the association file and tap the corresponding link on the device, the app does launch properly, and an additional log is generated:

`2016-07-05 15:38:51.174 HGDemo[2644:872425] IonicDeepLinkPlugin: Handle continueUserActivity (internal) https://<redacted>/public/#/ResetPassword/abc123`

For each of the tests above, I first verified the path payload for `apple-site-association` using this [universal links validator tool](https://branch.io/resources/universal-links/)

**Comment by theaccordance** [comment]
I've spent most of today digging into this further and I'm starting to suspect that the issue is related to iOS itself.

I modified the path in my `apple-app-site-association` file to use a single-character wildcard (`?`) in place of the hashtag.  I thought maybe iOS was having issues parsing the path defined in the association file:

```
{
    "applinks": {
        "apps": [],
        "details": [
            {
                "appID": "95FLFXC8CQ.com.highground.demo",
                "paths": [
                    "/public/?/ResetPassword/*"
                    // originally "public/#/ResetPassword/*"
                ]
            }
        ]
    }
}
```

So I deployed the updated file to my test server and proceeded to test my link (`https://<redacted>/public/#/ResetPassword/abc123`) without success. 

I then had the wild idea of swapping out the hashtag for another character in the link itself (`https://<redacted>/public/a/ResetPassword/abc123`) , just to see if iOS was parsing the url properly, and guess what?  It opened.  

Talk about a god damn struggle.  I'm an absolute noob when it comes to submitting bug reports to Apple, but I'm in the process of letting them know about this.

**Comment by mlynch** [comment]
Thanks @theaccordance. If the app doesn't open at all from an external click, that means the issue is somewhere else (your association file, iOS, etc.). Once the app opens, if it's not navigating properly that's the plugin's issue. When the app opened, did it navigate properly for you? 

As for the # character, it starts a hash fragment so is often treated in a special way (hence the original issue I had to fix).

**Comment by theaccordance** [comment]
Yeah that's what I concluded at the end of last week, the native code for the plugin matched the implementation guidelines in Apple's docs. Thankfully, I was able to figure a way to hack the path in `apple-app-site-association` that would work for my particular use case.  I'm hoping Apple will acknowledge my bug report and patch the problem (if I'm correct in my assumption) by the time my org wants to expand on the types of links that are opened in-app, otherwise we'll have to go all-in. 

Thanks for taking the time to respond, I know you guys have a lot of moving parts over there.
